pages/02.Tutorials/11.object-storage-acls/docs.en.md

```diff
@@ -0,0 +1,166 @@
+        - docs
+---
+
+## Object Storage ACLs
+
+### Overview
+
+In this tutorial we will use the [Object Storage](../../04.Reference/05.object-storage/docs.en.md) to create private buckets and objects and further limit the access to these, by applying ACLs granting read-only/write rights. We will be using [s3cmd](http://s3tools.org/s3cmd) manage our resources.
+
+### Prerequisites
+
+* You know the basics of using the [OpenStack CLI-Tools](../../03.Howtos/02.openstack-cli/docs.en.md).
+* Environment variables are set, like shown in the [API-Access-Tutorial](../../02.Tutorials/02.api-access/docs.en.md).
+* You have created EC2 credentials for your OpenStack user to be able to use the [Object Storage](../../04.Reference/05.object-storage/docs.en.md).
+* You have installed and configured [s3cmd](http://s3tools.org/s3cmd) with your EC2 credentials.
+* You have at least 1 additional user with access to your OpenStack project and you know the name of this user.
+
+### Create initial buckets and objects
+
+Using your personal s3cmd configuration, we will create 2 buckets. `acl-read` representing the bucket to which your additional user will get read-only access and `acl-write` representing the bucket to which your additional user gets read-write access.
+
+```shell
+# .s3cfg-admin represents your personal s3cmd config containing your access/secret key pair.
+s3cmd -c .s3cfg-admin mb s3://acl-read
+s3cmd -c .s3cfg-admin mb s3://acl-write
+# Create a test.txt file
+echo "secret-text" > test.txt
+# Upload the file in both buckets
+s3cmd -c .s3cfg-admin put test.txt s3://acl-read/test.txt
+s3cmd -c .s3cfg-admin put test.txt s3://acl-write/test.txt
+# We will later use the test.txt object to confirm our applied ACLs are working
+```
+
+As we did not define any ACLs while creating the buckets/objects, the default private ACL will be used. In the current state all users, who have access to our OpenStack project (users who have created EC2 credentials for this project), have full control on our newly created buckets/objects.
+
+For an object to be accessable to an user, the object itself and the parent buckets have to have proper ACLs set.
+
+### Revoke default ACLs 
+
+In this step we will revoke the access to the buckets/objects for every user with access to our OpenStack project.
+
+*As this point it is important to understand that OpenStack EC2 credentials are bound to a user and a project. Thus if you are managing multiple OpenStack projects, there is more then 1 way to allow an user to be able to access your buckets/objects. You may allow the users EC2 credentials created for project A to access your data but at the same time revoke the access for the users EC2 credentials created for project B* 
+
+```shell
+# Get our OpenStack project ID
+MY_PROJECT_ID=$(openstack project list -f value -c ID)
+# Revoke access to anyone except ourself from your OpenStack project
+s3cmd -c .s3cfg-admin setacl --acl-revoke=full_control:${MY_PROJECT_ID} s3://acl-read
```

Comment:
I expected `--acl-revoke=full_control:g:${MY_PROJECT_ID}` here, why is there just a project id?

Reply:
As the default ACL does not take groups into account, we cannot remove it using the `g` (group prefix). It simply allows access for all users of the project. We may allow groups to access the bucket afterwards tho : 

```
# Revoke default ACL
s3cmd -c .s3cfg-admin setacl --acl-revoke=full_control:${MY_PROJECT_ID} s3://acl-read
# Grant group ACL
s3cmd -c .s3cfg-admin setacl --acl-grant=full_control:g:${MY_GROUP_ID}/${MY_PROJECT_ID} s3://acl-read
# Re-revoke default ACL which got added again
s3cmd -c .s3cfg-admin setacl --acl-revoke=full_control:${MY_PROJECT_ID} s3://acl-read
```